main.tf

```diff
@@ -80,3 +80,43 @@ resource "google_cloudbuildv2_connection" "cloudbuild_github_project_connection"
   }
   depends_on = [google_secret_manager_secret_iam_policy.policy]
 }
+
+resource "google_cloudbuild_trigger" "filename-trigger" {
+  location = "europe-west1"
+
+  repository_event_config {
+    // The 'repository' attribute should reference a google_cloudbuildv2_repository resource.
+    // Ensure this repository connection and repository resource are defined elsewhere in your Terraform.
+    repository = "projects/syntax-errors/locations/europe-west1/connections/ai-agent-github-connection/repositories/kwame-mintah-hugging-face-smolagents-playground"
+    push {
+      branch       = "cloudbuild-yaml"
+      invert_regex = false
+    }
```

Comment:
Need to change this branch to be `main`. And then add a [additional configuration](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudbuild_trigger#pull_request-1) for pull request builds:

```
    pull_request {
        branch = "^main$"
        invert_regex = false
        comment_control = "COMMENTS_ENABLED"
    }
```

Reply:
done

Reply:
My fault, meant add on-to what we currently have

```hcl
  pull_request {
    branch          = "^main$"
    invert_regex    = false
    comment_control = "COMMENTS_ENABLED"
  }

  push {
    branch       = "^main$"
    invert_regex = false
  }
```

The idea is that we only run a new build if we comment `/gcbrun` on open pull request, to save our free minutes. But will always have a new docker image built when we push to the main branch.